client/platform/desktop/frontend/store/settings.ts

```diff
@@ -1,15 +1,47 @@
 import Vue from 'vue';
-import Install, { ref } from '@vue/composition-api';
-import { ipcRenderer } from 'electron';
+import Install, { ref, computed } from '@vue/composition-api';
+import { ipcRenderer, remote } from 'electron';
 import { Settings } from 'platform/desktop/constants';
 import { cloneDeep } from 'lodash';
+import * as semver from 'semver';
 
 // TODO remove this: this won't be necessary in Vue 3
 Vue.use(Install);
 
 const SettingsKey = 'desktop.settings';
+const VersionKey = 'desktop.currentVersion';
 
 const settings = ref(null as Settings | null);
+const currentVersion = remote.app.getVersion();
+const knownVersion = ref(window.localStorage.getItem(VersionKey));
+
+/**
+ * upgradedVersion indicates that the currently launched instance
+ * is an upgrade since the last time the user acknowledged an instnace of dive.
+ */
+const upgradedVersion = computed(() => {
+  const known = knownVersion.value;
+  if (known === null) {
+    // all versions are greater than null
+    return currentVersion;
```

Comment:
This is minor, If you don't acknowledge this through dismissing and then go to a lower version number you won't get the downgrade warning message?

Reply:
Right.  I could do a different implementation with 2 localstorage variables: the currentVersion that gets set immediately on launch and the `acknowleged` variable which gets set true and then back to false every time the version changes.

It was somewhat of an arbitrary choice but this seemed simpler. If you think it's worth doing the other way, I can.

Reply:
I wouldn't.  This I think is a solution for a different and larger problem and that's upgrading and notifying the user about the upgrades and installing them in a consistent location.